test/get_previous_releases.py

```diff
@@ -133,20 +134,22 @@ def download_binary(tag, args) -> int:
 
     print('Fetching: {tarballUrl}'.format(tarballUrl=tarballUrl))
 
-    header, status = subprocess.Popen(
-        ['curl', '--head', tarballUrl], stdout=subprocess.PIPE).communicate()
-    if re.search("404 Not Found", header.decode("utf-8")):
-        print("Binary tag was not found")
-        return 1
-
-    curlCmds = [
-        ['curl', '--remote-name', tarballUrl]
-    ]
+    try:
+        response = urlopen(Request(tarballUrl, method='HEAD'))
+    except HTTPError as e:
+        if e.code == 404:
+            print("Binary tag was not found")
+            return 1
+        else:
+            raise
 
-    for cmd in curlCmds:
-        ret = subprocess.run(cmd).returncode
-        if ret:
-            return ret
+    try:
+        response = urlopen(tarballUrl)
```

Comment:
AFAIK there's a minor behaviour change here, where `urlopen` *will* follow redirects whereas `curl` won't usually

```shell
$ curl -I https://httpbin.org/absolute-redirect/3
HTTP/2 302
# ...
```

```python
>>> from urllib.request import urlopen
>>> response = urlopen("https://httpbin.org/absolute-redirect/3")
>>> response.code
200  # Not 302 because redirects were followed
```

This should be fine, but worth a mention.

Reply:
Great point, Thank you.